File: angular/src/app/models/ville.ts
import { Actualite } from "./actualite";
import { Animal } from "./animal";

export class Ville {
    public get actualites(): Actualite[] {
        return this._actualites;
    }
    public set actualites(value: Actualite[]) {
        this._actualites = value;
    }
    public get animaux(): Animal[] {
        return this._animaux;
    }
    public set animaux(value: Animal[]) {
        this._animaux = value;
    }
    public get ambiance(): string {
        return this._ambiance;
    }
    public set ambiance(value: string) {
        this._ambiance = value;
    }
    public get nom(): string {
        return this._nom;
    }
    public set nom(value: string) {
        this._nom = value;
    }
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }
    constructor(
        private _id: number,
        private _nom: string,
        private _ambiance: string,
        private _animaux: Animal[],
        private _actualites: Actualite[]
    ) { }
}

File: spring/src/main/java/fr/formation/model/Ville.java
package fr.formation.model;

import java.util.List;

import org.springframework.beans.BeanUtils;

import com.fasterxml.jackson.annotation.JsonIgnore;

import fr.formation.api.response.VilleResponse;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;

@Entity
@Table(name = "ville")
public class Ville {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "vill_id")
	private int id;

	@NotBlank
	@Column(name = "vill_nom", length = 30, nullable = false)
	private String nom;

	@OneToMany(mappedBy = "ville")
	@JsonIgnore
	private List<Animal> animaux;

	@OneToMany(mappedBy = "ville")
	private List<Actualite> actualites;

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public List<Animal> getAnimaux() {
		return animaux;
	}

	public void setAnimaux(List<Animal> animaux) {
		this.animaux = animaux;
	}

	public List<Actualite> getActualites() {
		return actualites;
	}

	public void setActualites(List<Actualite> actualites) {
		this.actualites = actualites;
	}

	public VilleResponse convert() {
		VilleResponse villeResponse = new VilleResponse();
		BeanUtils.copyProperties(this, villeResponse);
		return villeResponse;
	}
}
